x-pack/plugins/infra/public/components/waffle/group_name.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { EuiLink, EuiToolTip } from '@elastic/eui';
+import React from 'react';
+import styled from 'styled-components';
+import { InfraWaffleMapGroup } from '../../lib/lib';
+
+interface Props {
+  onDrilldown: () => void;
+  group: InfraWaffleMapGroup;
+  isChild?: boolean;
+}
+
+export class GroupName extends React.PureComponent<Props, {}> {
+  public render() {
+    const { group, isChild } = this.props;
+    const linkStyle = {
+      fontSize: isChild ? '0.85em' : '1em',
+    };
+    return (
+      <GroupNameContainer>
+        <Inner isChild={isChild}>
+          <Name>
+            <EuiToolTip position="top" content={group.name}>
+              <EuiLink style={linkStyle} onClickCapture={this.handleClick}>
+                {group.name}
+              </EuiLink>
+            </EuiToolTip>
+          </Name>
+          <Count>{group.count}</Count>
+        </Inner>
+      </GroupNameContainer>
+    );
+  }
+
+  private handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    // TODO: fill this in with group click handler
+    event.preventDefault();
+    this.props.onDrilldown();
+  };
+}
+
+export const GroupNameContainer = styled.div`
+  position: relative;
+  text-align: center
+  font-size: 16px;
+  margin-bottom: 5px; 
+  top: 20px;
+  display: flex;
+  justify-content: center;
+  padding: 0 10px;
+`;
+
+interface InnerProps {
+  isChild?: boolean;
+}
+
+export const Inner = styled<InnerProps, 'div'>('div')`
+  border: 1px solid ${props => props.theme.eui.euiBorderColor};
+  background-color: ${props =>
```

Comment:
Can we maybe work with transparency here so each group does not have to know at which level it is? That way we can support deeper nesting and avoid one more prop from having to be passed down.

Reply:
This wasn't an issue in the prototype because we were using actual CSS to determine how to style nested groups vs. un-nested groups. I'm not sure what you mean by transparency. Are you talking about the `transparency` css directive?

Reply:
I mean using a alpha value < 1 for the color so the darker grays automatically build up with each nesting layer.

Reply:
Right... the issue with that is the label background needs to be solid because of how it overlaps the group container. I'm using alpha on the group container to achieve the same effect but if I did it on the `GroupName` the line for the group container would show through.

![image](https://user-images.githubusercontent.com/41702/43915969-1c631498-9bc1-11e8-9fbb-b2f3c5765b51.png)

Reply:
oh, I thought this was for the container... did not even see that the title had a different background :see_no_evil: makes sense, then